vm/src/frame.rs

```diff
@@ -264,6 +264,23 @@ impl FrameRef {
             self.lasti.get()
         }
     }
+
+    pub fn is_internal_frame(&self) -> bool {
+        let code = self.clone().f_code();
+        let filename = code.co_filename();
+
+        filename.as_str().contains("importlib") && filename.as_str().contains("_bootstrap")
+    }
+
+    pub fn next_external_frame(&self, vm: &VirtualMachine) -> Option<FrameRef> {
+        self.clone().f_back(vm).map(|back| {
+            if back.is_internal_frame() {
+                back
+            } else {
+                back.f_back(vm).unwrap()
+            }
```

Comment:
The function name and its behavior looks confusing.
The name is `next_external_frame` but the condition is `back.is_internal_frame()` than return the value.
If this is a correct implementation, we'd better to add comment how this is working.